### PR TITLE
Facade blind channel definitions and configuration structs

### PR DESCRIPTION
**Pull request by klew**

### CODE REVIEW

File: supla-common/proto.h [hunk 1]
```diff
@@ -412,6 +413,7 @@ extern char sproto_tag[SUPLA_TAG_SIZE];
 #define SUPLA_BIT_FUNC_RAINSENSOR 0x00002000                // ver. >= 12
 #define SUPLA_BIT_FUNC_WEIGHTSENSOR 0x00004000              // ver. >= 12
 #define SUPLA_BIT_FUNC_CONTROLLINGTHEROOFWINDOW 0x00008000  // ver. >= 13
+#define SUPLA_BIT_FUNC_CONTROLLINGTHEFACADEBLIND 0x00010000  // ver. >= 17
```
Comment (przemyslawzygmunt) [inline supla-common/proto.h:416]: I prefer the format 0x0010000

File: supla-common/proto.h [hunk 2]
```diff
@@ -1320,12 +1322,64 @@ typedef struct {
...
+// Device -> Server -> Client
+typedef struct {
+  signed char position;  // -1 == calibration. -1 - 100%, DSC
+  char reserved1;
+  signed char bottom_position;  // Percentage points to the windowsill, SC
+  _supla_int16_t flags;  // DSC
+  char reserved2;
+  char reserved3;
+  char reserved4;
+} TDSC_RollerShutterValue;
```
Comment (przemyslawzygmunt) [inline supla-common/proto.h:1339]: All new structures must be concreted here:
https://github.com/SUPLA/supla-core/blob/develop/supla-common/test/ProtoTest.cpp#L211
ASSERT_EX(sizeof(TDSC_RollerShutterValue), (unsigned int)X);

Structures which are placed in "value" should be verified if they are less than SUPLA_CHANNELVALUE_SIZE
ASSERT_LE(sizeof(TDSC_RollerShutterValue), (unsigned int)SUPLA_CHANNELVALUE_SIZE);

Structures which are placed in "Config" should be verified if they are less than SUPLA_CHANNEL_CONFIG_MAXSIZE
  ASSERT_LE(sizeof(TSD_ChannelConfig_FacadeBlind),
            (unsigned int)SUPLA_CHANNEL_CONFIG_MAXSIZE);

**Review by przemyslawzygmunt — CHANGES_REQUESTED** [review]

File: supla-common/proto.h (hunk 1 above)
Comment (klew) [reply supla-common/proto.h:416]: Current format is aligned with other BIT_FUNC entries above. Are you sure? :)
If yes, would you like me to align all above BIT_FUNC as well to have one digit less?

File: supla-common/proto.h [hunk 3]
```diff
@@ -1320,12 +1322,64 @@ typedef struct {
...
 #define RS_VALUE_FLAG_CALIBRATION_IN_PROGRESS 0x10
 
-typedef struct {
-  char position;  // -1 == calibration. -1 - 100%
-  char tilt;
-  char bottom_position;  // Percentage points to the windowsill
-  _supla_int16_t flags;
-} TRollerShutterValue;
+#define SUPLA_FACADEBLIND_TYPE_KEEP_POSITION_DURING_TILTING 1
+#define SUPLA_FACADEBLIND_TYPE_CHANGE_POSITION_DURING_TILTING 2
```
[the remark this answers is not in the document]
Comment (klew) [reply supla-common/proto.h:1326]: Changed to: SUPLA_FACADEBLIND_TYPE_CHANGES_POSITION_WHILE_TILTING

File: supla-common/proto.h [hunk 4]
```diff
@@ -1320,12 +1322,64 @@ typedef struct {
 #define RS_VALUE_FLAG_MOTOR_PROBLEM 0x8
 #define RS_VALUE_FLAG_CALIBRATION_IN_PROGRESS 0x10
 
-typedef struct {
-  char position;  // -1 == calibration. -1 - 100%
-  char tilt;
-  char bottom_position;  // Percentage points to the windowsill
-  _supla_int16_t flags;
-} TRollerShutterValue;
+#define SUPLA_FACADEBLIND_TYPE_KEEP_POSITION_DURING_TILTING 1
```
[the remark this answers is not in the document]
Comment (klew) [reply supla-common/proto.h:1325]: Changed to: SUPLA_FACADEBLIND_TYPE_STANDS_IN_POSITION_WHILE_TILTING

File: supla-common/proto.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (klew) [reply supla-common/proto.h:1326]: Changed to: SUPLA_FACADEBLINE_TYPE_TILTS_ONLY_WHEN_FULLY_CLOSED

File: supla-common/proto.h [hunk 5]
```diff
@@ -1320,12 +1322,64 @@ typedef struct {
...
+  signed char position;  // -1 == calibration. -1 - 100%, DSC
+  char reserved1;
+  signed char bottom_position;  // Percentage points to the windowsill, SC
+  _supla_int16_t flags;  // DSC
+  char reserved2;
+  char reserved3;
+  char reserved4;
+} TDSC_RollerShutterValue;
+
+typedef TDSC_RollerShutterValue TRollerShutterValue;
```
[the remark this answers is not in the document]
Comment (klew) [reply supla-common/proto.h]: done

**Pull request merged**